src/main/java/io/confluent/connect/jdbc/sink/dialect/HANADialect.java

```diff
@@ -0,0 +1,90 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.confluent.connect.jdbc.sink.dialect;
+
+import io.confluent.connect.jdbc.sink.metadata.SinkRecordField;
+import org.apache.kafka.connect.data.Schema;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static io.confluent.connect.jdbc.sink.dialect.StringBuilderUtil.joinToBuilder;
+import static io.confluent.connect.jdbc.sink.dialect.StringBuilderUtil.nCopiesToBuilder;
+
+public class HANADialect extends DbDialect {
+  private static final Logger log = LoggerFactory.getLogger(HANADialect.class);
+
+  public HANADialect() {
+    super(getSqlTypeMap(), "\"", "\"");
+  }
+
+  private static Map<Schema.Type, String> getSqlTypeMap() {
+    log.trace("Map to HANA Dialect");
+    Map<Schema.Type, String> map = new HashMap<>();
+    map.put(Schema.Type.INT8, "TINYINT");
+    map.put(Schema.Type.INT16, "SMALLINT");
+    map.put(Schema.Type.INT32, "INTEGER");
+    map.put(Schema.Type.INT64, "BIGINT");
+    map.put(Schema.Type.FLOAT32, "REAL");
+    map.put(Schema.Type.FLOAT64, "DOUBLE");
+    map.put(Schema.Type.BOOLEAN, "BOOLEAN");
+    map.put(Schema.Type.STRING, "VARCHAR(1000)");
+    map.put(Schema.Type.BYTES, "BLOB");
+    return map;
+  }
+
+  @Override
+  public String getCreateQuery(String tableName, Collection<SinkRecordField> fields) {
+
+    // Defaulting to Column Stores
+    String createQuery = super.getCreateQuery(tableName, fields).replace("CREATE TABLE", "CREATE COLUMN TABLE");
+    log.trace("Create Query : " + createQuery);
+    return createQuery;
+  }
+
+  @Override
+  public List<String> getAlterTable(String tableName, Collection<SinkRecordField> fields) {
+    final StringBuilder builder = new StringBuilder("ALTER TABLE ");
+    builder.append(escaped(tableName));
+    builder.append(" ADD(");
+    writeColumnsSpec(builder, fields);
+    builder.append(")");
+    log.trace("Alter Table query: " + builder.toString());
+    return Collections.singletonList(builder.toString());
+  }
+
+  @Override
+  public String getUpsertQuery(final String table, Collection<String> keyCols, Collection<String> cols) {
+      // https://help.sap.com/hana_one/html/sql_replace_upsert.html
+
+    StringBuilder builder = new StringBuilder("UPSERT ");
+    builder.append(escaped(table));
+    builder.append("(");
+    joinToBuilder(builder, ",", keyCols, cols, escaper());
+    builder.append(") VALUES(");
+    nCopiesToBuilder(builder, ",", "?", keyCols.size() + cols.size());
+    builder.append(")");
+    builder.append(" WITH PRIMARY KEY");
+    log.trace("Upsert Query: " + builder.toString());
```

Comment:
we log this already for each batch, in `BufferedRecords.add()`, so let's remove this log line